Add toast notification tray

peppy opened the pull request:
PRing this ahead of fixing existing tests to make sure everyone is on board with the UX.

- [x] Depends on #20030.
- Closes #18885
- Closes #10222
- Closes #5901

https://user-images.githubusercontent.com/191335/187437527-6b9a3d65-4306-4d4b-86df-defa123257b2.mp4

peppy committed:
Add flash when a new notification is displayed to draw attention

peppy committed:
Add toast notification tray

peppy committed:
Change global overlay ordering so notification toasts display above settings

peppy committed:
Reduce notification post delay now that it's less important

peppy committed:
Play notification appear sample immediately

peppy committed:
Fix notification completion events not being run when overlay not visible

peppy committed:
Don't dismiss toasts while hovered (and adjust timings slightly)

peppy committed:
Improve forwarding flow to not use piling delegates

peppy committed:
Remove notification blocking behaviour of first run setup

peppy committed:
Update remaining test expectations with new behaviour

peppy committed:
Fix remaining test failures by strengthening `PlayerLoader` tests

- Click using `TriggerClick` as notifications move around quite a bit.
- Ensure any notifications from a previous test method are cleaned up.

bdach commented:
On a quick look I think this looks pretty good. If I were to gripe then I'd say maybe the flash is a bit bright but probably good to go with for a first revision anyway.

peppy committed:
Mark toasts as read when closing the overlay for added safety

I'm not sure how the read status will play out going forward so I'm just
adding this to keep things conforming for now.

peppy committed:
Fix unread count potentially missing notifications in a transforming state

peppy committed:
Fix `IntroTestScene` not clearing previous notifications hard enough

smoogipoo committed:
Merge branch 'master' into toast-notification-tray

smoogipoo reviewed (approved):
Seems ok
